src/zenml/zen_stores/schemas/artifact_schemas.py

```diff
@@ -160,9 +162,16 @@ def to_model(self) -> ArtifactResponseModel:
             else:
                 producer_step_run_id = step_run.original_step_run_id
 
+        version: Union[str, int]
+        try:
+            version = int(self.version)
+        except ValueError:
+            version = self.version
```

Comment:
Does it change anything if you will not cast numerics to int here?

Reply:
No actually not. I think I'll just keep them as string to make the typing easier so we don't always need `Union` everywhere